src/drivers/uefi/lv_uefi_indev_keyboard.c

```diff
@@ -0,0 +1,349 @@
+    if(!_simple_text_input_interface_is_valid(interface)) {
+        lv_uefi_protocol_close(handle, &_uefi_guid_simple_text_input);
+        LV_LOG_WARN("[lv_uefi] The SIMPLE_TEXT_INPUT interface is not valid.");
+        return false;
+    }
+
+    handle_ctx = (lv_uefi_simple_text_input_handle_context_t *) lv_ll_ins_head(&indev_ctx->handles);
+    LV_ASSERT_MALLOC(handle_ctx);
+
+    handle_ctx->handle = handle;
+    handle_ctx->interface = interface;
+
+    return true;
+}
+
+/**
+ * @brief Add all available EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL interfaces to the indev.
+ * @param indev Indev that was created with lv_uefi_simple_text_input_indev_create.
+*/
+void lv_uefi_simple_text_input_indev_add_all(
+    lv_indev_t * indev)
+{
+    EFI_STATUS status;
+    EFI_HANDLE * handles = NULL;
+    UINTN no_handles;
+    UINTN index;
+
+    lv_uefi_simple_text_input_context_t * indev_ctx = (lv_uefi_simple_text_input_context_t *)lv_indev_get_user_data(indev);
+    LV_ASSERT_NULL(indev_ctx);
+
+    if(indev_ctx->signature != SIMPLE_TEXT_INPUT_INDEV_SIGNATURE) return;
+
+    status = gLvEfiBS->LocateHandleBuffer(ByProtocol, &_uefi_guid_simple_text_input, NULL, &no_handles, &handles);
+    if(status != EFI_SUCCESS) {
+        LV_LOG_ERROR("[lv_uefi] LocateHandleBuffer(SIMPLE_TEXT_INPUT_EX) failed with error code %llx.", status);
+        return;
+    }
+
+    for(index = 0; index < no_handles; index++) {
+        lv_uefi_simple_text_input_indev_add_handle(indev, handles[index]);
+    }
+
+    if(handles != NULL) gLvEfiBS->FreePool(handles);
+}
+
+/**********************
+ *   STATIC FUNCTIONS
+ **********************/
+
+static void _simple_text_input_event_cb(lv_event_t * e)
+{
+    lv_indev_t * indev;
+    lv_uefi_simple_text_input_context_t * indev_ctx;
+
+    if(lv_event_get_code(e) != LV_EVENT_DELETE) return;
+
+    indev = (lv_indev_t *)lv_event_get_user_data(e);
+    if(indev == NULL) return;
+
+    indev_ctx = (lv_uefi_simple_text_input_context_t *)lv_indev_get_user_data(indev);
+    lv_indev_set_user_data(indev, NULL);
+
+    if(indev_ctx != NULL) _simple_text_input_context_free(indev_ctx);
+}
+
+static void _simple_text_input_read_cb(lv_indev_t * indev, lv_indev_data_t * data)
+{
+    lv_uefi_simple_text_input_handle_context_t * handle_ctx = NULL;
+    lv_uefi_simple_text_input_key_cache_t * key_cache = NULL;
+    void * node = NULL;
+
+    lv_uefi_simple_text_input_context_t * indev_ctx = (lv_uefi_simple_text_input_context_t *)lv_indev_get_user_data(indev);
+    LV_ASSERT_NULL(indev_ctx);
+
+    // Empty the buffer before reading new values
+    if(lv_ll_is_empty(&indev_ctx->key_cache)) {
+        // Read from all registered devices
+        for(node = lv_ll_get_head(&indev_ctx->handles); node != NULL; node = lv_ll_get_next(&indev_ctx->handles, node)) {
+            handle_ctx = (lv_uefi_simple_text_input_handle_context_t *) node;
+            _simple_text_input_read(indev_ctx, handle_ctx);
+        }
+    }
+
+    // Return the first value
```

Comment:
```suggestion
    /* Return the first value */
```

and for the others too.